### ISSUE TITLE
Make this the reference implementation

### ISSUE THREAD

**Issue by boronine** [husl-colors/husl-haxe#7]
@Eiyeron I have good news. https://github.com/husl-colors/husl-haxe/tree/js-target

This branch generates a file `husl.min.js` which is a drop-in replacement for `husl.js` in https://github.com/husl-colors/husl. It passes the reference test suite. I used closure-compiler to optimize and minify the code, and performance wise it seems to be identical to the original implementation. I definitely want to make this the reference implementation.

I would prefer to merge this repo (history intact) into https://github.com/husl-colors/husl.

Is the following okay for copyright notice? I'll put it in the central `LICENSE` file.

```
Copyright (c) 2012-2016 Alexei Boronine
Copyright (c) 2016 Florian Dormont

Permission is hereby granted, free of charge, to any person obtaining a copy ...
```

If you have work-in-progress in this repo, you should push it now.

**Comment by Eiyeron** [comment]
It's okay for the licence notice. I'll close the issue when I'll be done with the documentation.

**Comment by boronine** [comment]
👍

**Comment by Eiyeron** [comment]
You might also want to check out `@export` in Haxe [there](https://haxe.org/manual/target-javascript-expose.html). I don't know if it matches what you wanted to do but it allows external scripts to use Haxe generated classes like normal JS classes.

Also I finally pushed my changes, there was a bizarre issue where it looked like I was missing parts of the commit history, making pushing impossible. I squashed my changes into the last two commits. I'm not sure about Husl.hx documentation if it looks correct or not.
